client/checkout/blocks/payment-elements.js

```diff
@@ -66,41 +78,43 @@ const PaymentElements = ( { api, ...props } ) => {
 	}
 
 	return (
-		<LoadableBlock isLoading={ ! appearance } numLines={ 3 }>
-			<Elements
-				stripe={ stripeForUPE }
-				options={ {
-					mode: amount < 1 ? 'setup' : 'payment',
-					amount: amount,
-					currency: currency,
-					paymentMethodCreation: 'manual',
-					paymentMethodTypes: paymentMethodTypes,
-					appearance: appearance,
-					fonts: fontRules,
-				} }
-			>
-				{ paymentProcessorLoadErrorMessage?.error?.message && (
-					<div className="wc-block-components-notices">
-						<StoreNotice status="error" isDismissible={ false }>
-							<RawHTML>
-								{
-									paymentProcessorLoadErrorMessage.error
-										.message
-								}
-							</RawHTML>
-						</StoreNotice>
-					</div>
-				) }
-				<PaymentProcessor
-					api={ api }
-					errorMessage={ errorMessage }
-					fingerprint={ fingerprint }
-					onLoadError={ setPaymentProcessorLoadErrorMessage }
-					theme={ appearance?.theme }
-					{ ...props }
-				/>
-			</Elements>
-		</LoadableBlock>
+		<div ref={ containerRef }>
+			<LoadableBlock isLoading={ ! appearance } numLines={ 3 }>
```

Comment:
I suppose that it's a micro-optimization now, but what do you think of changing this line to
```suggestion
			<LoadableBlock isLoading={ ! appearance || !stripeForUPE } numLines={ 3 }>
```

and removing the `if ( ! stripeForUPE )` check a few lines above?

I'm thinking that with your changes, the `generateUPEAppearance()` function won't be executed until `stripeForUPE` has been provided.
But if instead we move the conditional on `stripeForUPE` below, we can load Stripe _and_ generate the UPE appearance concurrently.